MekHQ/src/mekhq/campaign/finances/Loan.java

```diff
@@ -93,102 +87,72 @@ public Loan(Money p, int r, int c, int y, int s, GregorianCalendar today, String
     }
 
     public void setFirstPaymentDate() {
-        boolean keepGoing = true;
         //We are going to assume a standard grace period, so you have to go
         //through the first full time length (not partial) before your first
         //payment
-        int grace = 1;
-        switch(schedule) {
-        case Finances.SCHEDULE_BIWEEKLY:
-            grace = 2;
-            break;
-        case Finances.SCHEDULE_QUARTERLY:
-            grace = 3;
-            break;
-        }
-        while(keepGoing) {
-            nextPayment.add(GregorianCalendar.DAY_OF_YEAR, 1);
-            switch(schedule) {
+
+        // First, we need to increase the number of days by one
+        nextPayment = nextPayment.plusDays(1);
+
+        // Then, we create a variable to hold temporary difference values
+        int temp;
```

Comment:
Still needed?

Reply:
Nope, and odd that my IDE didn't prompt about it being unused.